File: DebriefRepository/src/main/java/org/example/debriefrepository/entity/Role.java
package org.example.debriefrepository.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "role", schema = "debrief_mgmt")
public class Role extends BaseEntity {

    @Column(name = "name", nullable = false)
    private String name;

    @OneToMany(mappedBy = "role")
    private List<UserRole> users = new ArrayList<>();

}

File: DebriefRepository/src/main/java/org/example/debriefrepository/entities/Role.kt
package org.example.debriefrepository.entities

import jakarta.persistence.*

@Entity
@Table(name = "role", schema = "debrief_mgmt")
open class Role {
    @Id
    @Column(name = "id", nullable = false)
    open var id: Short? = null

    @Column(name = "role_name", nullable = false, length = Integer.MAX_VALUE)
    open var roleName: String? = null

    @ManyToMany
    @JoinTable(
        name = "user_roles",
        joinColumns = [JoinColumn(name = "role_id")],
        inverseJoinColumns = [JoinColumn(name = "user_id")]
    )
    open var users: MutableSet<org.example.debriefrepository.entities.User> = mutableSetOf()
}